### PR TITLE
feat(theme): add gridLineStyle to AxisConfig

### PR DESCRIPTION
**Pull request by rshen91**
## Summary
Fixes https://github.com/elastic/elastic-charts/issues/237

This PR will simplify the process of changing styles by moving gridLineStyle into the Theme interface through the AxisConfig interface. 

Light theme and dark theme are changed to include gridLineStyle.

BREAKING CHANGE: Added `GridLineStyle` to `Theme` (`theme.gridLineStyle.horizontal` and `theme.gridLineStyle.vertical`)

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [X] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [x] Proper documentation or storybook story was added for features that require explanation or tutorials
- [x] Unit tests were updated or added to match the most common scenarios
- [X] Each commit follows the [convention](https://github.com/elastic/elastic-charts/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by elasticcla** [comment]
Hi @rshen91, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in your Git [commit](https://github.com/elastic/elastic-charts/pull/257.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by rshen91** [comment]
Zoomed with Emma about this PR: 

- Need to define a theme fallback if axisSpec isn't specified by the user but can keep axisSpec in reactiveChart working as it was previously
- AxisSpec should take precedence over the theme but if a prop is defined, use that instead
- Define a helper function, in reactive_chart, that will handle the merge of the theme and the prop specified 
- Add a story to show the different input similar to the styling story called custom series style bars - apply bar style (bar 1 series) defaults to the theme if the customization isn't specified

File: src/lib/themes/dark_theme.ts [hunk 1]
```diff
@@ -87,6 +87,12 @@ export const DARK_THEME: Theme = {
       stroke: 'white',
       strokeWidth: 1,
     },
+    gridLineStyle: {
+      stroke: 'black',
+      strokeWidth: 1,
+      opacity: 0,
+      dash: [0, 0],
+    },
```
Comment (markov00) [inline src/lib/themes/dark_theme.ts]: What if we add also the `visible:true/false` so on the theme we don't have to work on the opacity to show or hide them?
I would also propose to set some nice default, like a darkgray or lightgray stroke for the dark theme  grid and the same/similar for the light theme

File: src/components/react_canvas/grid.tsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { Dimensions } from '../../lib/utils/dimensions';
 interface GridProps {
   chartDimensions: Dimensions;
   debug: boolean;
-  gridLineStyle: GridLineConfig | undefined;
+  gridLineStyle?: GridLineConfig;
```
Comment (markov00) [inline src/components/react_canvas/grid.tsx]: As this will come from the theme, it will always exist, so you can remove the optional parameter

**Comment by cchaos** [comment]
I'm sorry to throw a wrench in this, but can I also request that there are 2 grid line styles in the theme? One for horizontal and one for vertical. Here you can see that the EUI theme uses 2 different styles for each:

**Comment by rshen91** [comment]
Zoom with Emma
- Add visible to the theme
- Horizontal and Vertical specification for gridLineStyle 
- Add a new story in grid.tsx to address this new functionality (specify part of the axisSpec.gridLineStyle and make sure the theme generates the other properties for the gridLineStyle) 
-- Full editing feature for all parts of the theme gridLineStyle and a knob (a slider) for one part of the axisSpec (the others will be undefined by default)

File: src/components/react_canvas/grid.tsx (hunk 2 above)
Comment (rshen91) [reply src/components/react_canvas/grid.tsx]: Thank you! I think commit 79c2d11 addresses this

File: src/lib/themes/dark_theme.ts (hunk 1 above)
Comment (rshen91) [reply src/lib/themes/dark_theme.ts]: added visible here 3e55aec

**Comment by cchaos** [comment]
Any movement on this one? I'd **really** like to have it! ❤️

**Comment by nickofthyme** [comment]
@markov00 I think all the comments have been addressed.

**Pull request merged**